WordPress/Classes/Utility/WPURLRequest.m

```diff
@@ -22,6 +22,12 @@ + (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
     NSParameterAssert(password != nil || bearerToken != nil);
     
     NSMutableURLRequest *request = [self mutableRequestWithURL:loginUrl userAgent:userAgent];
+
+    NSString *hostname = [loginUrl host];
+    if (![hostname isEqualToString:@"wordpress.com"] && ![hostname hasSuffix:@".wordpress.com"]) {
```

Comment:
You can use [this helper](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Categories/NSString%2BHelpers.m#L191) instead